internal/controller/datadogagent/defaults/datadogagent_default.go

```diff
@@ -273,11 +274,20 @@ func defaultFeaturesConfig(ddaSpec *v2alpha1.DatadogAgentSpec) {
 	}
 	apiutils.DefaultBooleanIfUnset(&ddaSpec.Features.EBPFCheck.Enabled, defaultEBPFCheckEnabled)
 
+	// Service Discovery Feature
 	if ddaSpec.Features.ServiceDiscovery == nil {
 		ddaSpec.Features.ServiceDiscovery = &v2alpha1.ServiceDiscoveryFeatureConfig{}
 	}
 	apiutils.DefaultBooleanIfUnset(&ddaSpec.Features.ServiceDiscovery.Enabled, defaultServiceDiscoveryEnabled)
 
+	if *ddaSpec.Features.ServiceDiscovery.Enabled {
+		if ddaSpec.Features.ServiceDiscovery.NetworkStats == nil {
+			ddaSpec.Features.ServiceDiscovery.NetworkStats = &v2alpha1.ServiceDiscoveryNetworkStatsConfig{}
+		}
+
+		apiutils.DefaultBooleanIfUnset(&ddaSpec.Features.ServiceDiscovery.NetworkStats.Enabled, defaultServiceDiscoveryNetworkStatsEnabled)
```

Comment:
>  If not specified, the feature should be enabled, if service discovery itself is enabled

In my testing, if `features.serviceDiscovery.enabled=true`, the `DD_DISCOVERY_NETWORK_STATS_ENABLED` env var is added to the system-probe container, but defaults to `false`. That seems contrary to the described expected behavior